docs/api_proposal.md

```diff
@@ -0,0 +1,146 @@
+# Taxonomy API proposal
+
+Design document for DINA taxonomy API methods. Expanded from [Taxonomy API discussion](https://docs.google.com/spreadsheets/d/1uaJh1qt6mvY0ZCEB6uBKdnUnyGdpQnZcnFqTMHTS5pA/edit#gid=0) google doc.
+
+In this doc:
+* [Assumptions](#assumptions)
+* [Methods](#methods)
+* [Questions](#questions)
+
+See [DINA glossary](https://github.com/DINA-Web/dina-use-cases/blob/master/glossary.md) for terms.
+
+See [API example doc](https://github.com/DINA-Web/taxonomy/blob/master/docs/api_examples.md) for notes on non-DINA taxonomy APIs
+
+## Assumptions
+
+* taxonomy = hierarchy = classification = checklist
+* there will be some canonical root URL for the API, format to be determined
+* the API will be versioned, and we may deprecate older versions in the future
+* exact format / fields of result TBD; examples below are just for discussion about general form of result for each method
+* taxonomic names are not guaranteed to be unique
+  * therefore, cannot safely perform CRUD options based on name strings without disambiguation
+* one goal is to harmonize with existing taxonomy APIs (improved interoperability with other services and future modules)
```

Comment:
returning Dwc terms (when possible) may also improve interoperability

Reply:
+"Defined end points may be split or joined TBD + format/fields of GET, etc parameters TBD
+"Final PAI conformant with https://github.com/DINA-Web/guidelines/blob/master/DINA-Web-API-Guidelines.md